lib/StripeResource.js

```diff
@@ -102,7 +102,59 @@ StripeResource.prototype = {
     };
   },
 
-  _responseHandler(req, callback) {
+  _addHeadersDirectlyToResponse(res, headers) {
+    // For convenience, make some headers easily accessible on
+    // lastResponse.
+    res.requestId = headers['request-id'];
+    res.stripeAccount = res.stripeAccount || headers['stripe-account'];
+    res.apiVersion = res.apiVersion || headers['stripe-version'];
+    res.idempotencyKey = res.idempotencyKey || headers['idempotency-key'];
+  },
+
+  _makeResponseEvent(req, res, headers) {
+    const requestEndTime = Date.now();
+    const requestDurationMs = requestEndTime - req._requestStart;
+
+    return utils.removeNullish({
+      api_version: headers['stripe-version'],
+      account: headers['stripe-account'],
+      idempotency_key: headers['idempotency-key'],
+      method: req._requestEvent.method,
+      path: req._requestEvent.path,
+      status: res.statusCode,
+      request_id: res.requestId,
+      elapsed: requestDurationMs,
+      request_start_time: req._requestStart,
+      request_end_time: requestEndTime,
+    });
+  },
+
+  /**
+   * Used by methods with spec.streaming === true. For these methods, we do not buffer the
+   * response into memory, or do any parsing into stripe objects or errors, we
+   * delegate that all of that to the user and pass back the raw http.Response
+   * object to the callback.
+   */
```

Comment:
This threw me off a bit due to the mention of errors. I'd potentially clarify that the errors should be handled before this?

Reply:
Great catch